### DISCUSSION TITLE
Can I connect to databases with path of driver files on disk instead of installed drivers on Windows?

### DISCUSSION THREAD

**Discussion by laferce** [mkleehammer/pyodbc#1200] [Q&A] [answered]
My project will connect to many types of database, and it may used on some pc which program installing is forbbiden. So I want to know  how can I connect to them (Excel, SQLite, Access, MySQL, SQL Server, Oracle and so on) without drivers installed, but with driver files in my project dirctory? (As I know, sqlalchemy can do it like this to most databases, but I can't easily connect to Access.)

**Comment by v-chojas** [comment]
On Windows, if you have write access to HKLM\Software\ODBC you may be able to add your driver registry keys there to point to the driver files and remove them upon exit. Otherwise I don't think ODBC is a good solution to your problem since it's intended as a system-wide subsystem.

On Linux/Mac, I believe unixODBC does allow specifying a driver by path instead of name.

**Reply by laferce** [reply, marked as answer]
I've figured out how to connect to Access using SQLAlchemy in the same way as other databases. Thank you.
